OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -48,61 +65,173 @@ public override Activity Tick(Actor self)
 
 			if (IsCanceling || target.Type == TargetType.Invalid)
 			{
-				aircraft.RemoveInfluence();
+				if (landingInitiated)
+					aircraft.RemoveInfluence();
+
 				return NextActivity;
 			}
 
+			// Reevaluate target position in case the target has moved.
+			targetPosition = this.target.CenterPosition + this.offset;
+			landingCell = self.World.Map.CellContaining(targetPosition);
+
+			// We are already at the landing location.
+			if ((targetPosition - aircraft.GetPosition()).Length == 0)
```

Comment:
Fixed by changing to `LengthSquared`. It definitely should not be `HorizontalLengthSquared` because that would mean the aircraft would be content hovering above the landing spot instead of actually landing.